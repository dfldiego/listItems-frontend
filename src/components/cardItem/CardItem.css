.cardItem_container {
  background-color: var(--white);
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 1fr 1fr;
  margin: 1.6rem;
  cursor: pointer;
}
.cardItem_container:hover {
  transform: scale(1.05);
  box-shadow: 0px 0px 30px -15px var(--white-moderate);
  transition: all 2500ms ease;
}
@media (min-width: 768px) {
  .cardItem_container {
    background-color: var(--white);
    display: grid;
    grid-template-columns: 19.6rem 1fr;
    margin: 1.6rem;
    max-height: 18rem;
    cursor: pointer;
  }
  .cardItem_details {
    display: flex;
    justify-content: space-between;
    max-height: 18rem;
    padding: 1rem;
  }
}

/* IMAGEN */
.cardItem_img {
  width: 18rem;
  height: 18rem;
  margin-right: 1.6rem;
  border-radius: 0.4rem;
}
/* SECCION DERECHA DE LA CARD

/* PRECIO - SHIPPING */
.cardItem_amount_shipping {
  display: flex;
  align-items: center;
}
.free_shipping_img {
  width: 1.6rem;
  height: 1.6rem;
}
/* TITLE */
@media (min-width: 768px) {
  .title {
    font-size: 1.8rem;
  }
  .cardItem_title {
    padding: 3.2rem 0;
  }
}
.title {
  font-size: 1.4rem;
  width: 1rem;
}
.cardItem_title {
  padding: 1.6rem 0;
}
/* SELLER STATE */
@media (min-width: 768px) {
  .cardItem_sellerState {
    margin-top: 1.6rem;
    margin-right: 1.6rem;
    padding: 0 3.2rem;
  }
}
.cardItem_sellerState {
  padding: 0;
}
.cardItem_border {
  border-bottom: 1px solid var(--white-moderate);
}
