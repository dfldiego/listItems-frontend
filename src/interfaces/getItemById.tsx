export interface GetItemByIdResponse {
  data: DataChild;
}

export interface DataChild {
  ok: boolean;
  response: Response;
}

export interface Response {
  author: Author;
  item: Item;
}

export interface Author {
  nickname: null;
}

export interface Item {
  id: string;
  title: string;
  price: Price;
  picture: string;
  condition: string;
  free_shipping: boolean;
  sold_quantity: number;
  description: string;
}

export interface Price {
  currency: string;
  amount: number;
  decimals: number;
}

export interface useItemResponse {
  itemResponse: ProductDetail | null;
  error: string;
  loading: boolean;
}

export interface ProductDetail {
  author: Author;
  item: ItemDetail;
}

export interface ItemDetail {
  id: string;
  title: string;
  price: Price;
  picture: string;
  condition: string;
  free_shipping: boolean;
  sold_quantity: number;
  description: string;
}
