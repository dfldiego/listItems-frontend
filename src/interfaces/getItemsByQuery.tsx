export interface GetItemsByQueryResponse {
  data: DataChild;
}

export interface DataChild {
  ok: boolean;
  response: Response;
}

export interface Response {
  categories: any[];
  items: ItemElement[];
  totalPages: number;
}

export interface ItemElement {
  author: Author;
  item: ItemItem;
}

export interface Author {
  nickname: string;
}

export interface ItemItem {
  id: string;
  title: string;
  price: Price;
  picture: string;
  condition: string;
  free_shipping: boolean;
  sellerState: string;
}

export interface Price {
  currency: string;
  amount: number;
  decimals: number;
}
