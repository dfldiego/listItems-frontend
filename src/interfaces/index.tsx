export interface Item {
  author: {
    nickname?: string | null;
  };
  item: {
    id: string;
    title: string;
    price: {
      currency: string;
      amount: number;
      decimals: number;
    };
    picture: string;
    condition: string;
    free_shipping: boolean;
    sold_quantity?: number;
    description?: string;
    sellerState?: string;
  };
}

export interface ProductDetail {
  author: Author;
  item: ItemDetail;
}

export interface Author {
  nickname: null;
}

export interface ItemDetail {
  id: string;
  title: string;
  price: Price;
  picture: string;
  condition: string;
  free_shipping: boolean;
  sold_quantity: number;
  description: string;
}

export interface Price {
  currency: string;
  amount: number;
  decimals: number;
}

export interface ItemsState {
  breadcrumbs: string[] | [];
}
