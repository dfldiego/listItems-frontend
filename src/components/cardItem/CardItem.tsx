import "./CardItem.css";
import { Item } from "../../interfaces";
import ic_shipping from "../../assets/ic_shipping.png";
import { useNavigate } from "react-router-dom";
type Props = {
  item: Item;
};

const CardItem = ({ item }: Props) => {
  const navigate = useNavigate();
  const { author, item: itemElement } = item;
  const { id, picture, title, price, free_shipping, sellerState } = itemElement;

  const handleClick = (id: string) => {
    if (!id) return;
    navigate(`/items/${id}`);
  };

  return (
    <>
      <div
        className="cardItem_container"
        data-testid="card-container-test-id"
        onClick={() => handleClick(id)}
      >
        <img
          src={picture}
          alt={`${title}`}
          className="cardItem_img"
          data-testid="img-title-test-id"
        />
        <div className="cardItem_details">
          <div>
            <div className="cardItem_amount_shipping mt16">
              <div className="pr1">
                <span className="fontSize24">$ {price.amount}</span>
              </div>
              <div>
                {!free_shipping && (
                  <img
                    src={ic_shipping}
                    alt={"free_shipping"}
                    className="free_shipping_img"
                    data-testid="img-free-shipping-test-id"
                  />
                )}
              </div>
            </div>
            <div className="cardItem_title ">
              <span className="title">{title}</span>
            </div>
            <p className="fontSize12">{author.nickname}</p>
          </div>
          <div className="mt16 px32 cardItem_sellerState">
            <span className="fontSize12">{sellerState}</span>
          </div>
        </div>
      </div>

      <div className="cardItem_border"></div>
    </>
  );
};

export default CardItem;
