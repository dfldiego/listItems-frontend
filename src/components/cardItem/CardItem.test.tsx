import { fireEvent, render, screen } from "@testing-library/react";
import { Item } from "../../interfaces";
import CardItem from "./CardItem";

// MOCK useNavigate
const mockNavigate = jest.fn();
jest.mock("react-router-dom", () => ({
  useNavigate: () => mockNavigate,
}));

// MOCK PROPS ITEM
const mockItem: Item = {
  author: {
    nickname: "QUIRINO BEBIDAS",
  },
  item: {
    id: "MLA1308919460",
    title: "Aperitivo Aperol 750ml",
    price: {
      currency: "ARS",
      amount: 1298,
      decimals: 0,
    },
    picture: "http://http2.mlstatic.com/D_653025-MLA31576823324_072019-O.jpg",
    condition: "new",
    free_shipping: false,
    sellerState: "Capital Federal",
  },
};

describe("<CardItem>", () => {
  test("should render correctly", () => {
    const { container } = render(<CardItem item={mockItem} />);
    expect(container).toMatchSnapshot();
  });

  test("should render the props data", () => {
    render(<CardItem item={mockItem} />);
    // title
    expect(screen.getByText(/Aperitivo Aperol 750ml/i)).toBeInTheDocument();
    // amount
    expect(screen.getByText(/1298/i)).toBeInTheDocument();
    // nickname seller
    expect(screen.getByText(/QUIRINO BEBIDAS/i)).toBeInTheDocument();
    // sellerState
    expect(screen.getByText(/Capital Federal/i)).toBeInTheDocument();
    // imgs
    const imgTitle = screen.getByTestId("img-title-test-id");
    expect(imgTitle).toBeInTheDocument();
    const imgFreeShipping = screen.getByTestId("img-free-shipping-test-id");
    expect(imgFreeShipping).toBeInTheDocument();
  });

  test("I should navigate to the item detail page when I click on the item card.", () => {
    render(<CardItem item={mockItem} />);
    const card = screen.getByTestId("card-container-test-id");
    fireEvent.click(card);
    expect(mockNavigate).toHaveBeenCalled();
    expect(mockNavigate).toHaveBeenCalledWith(`/items/${mockItem.item.id}`);
  });
});
