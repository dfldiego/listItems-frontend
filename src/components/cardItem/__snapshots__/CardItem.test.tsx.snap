// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardItem> should render correctly 1`] = `
<div>
  <div
    class="cardItem_container"
    data-testid="card-container-test-id"
  >
    <img
      alt="Aperitivo Aperol 750ml"
      class="cardItem_img"
      data-testid="img-title-test-id"
      src="http://http2.mlstatic.com/D_653025-MLA31576823324_072019-O.jpg"
    />
    <div
      class="cardItem_details"
    >
      <div>
        <div
          class="cardItem_amount_shipping mt16"
        >
          <div
            class="pr1"
          >
            <span
              class="fontSize24"
            >
              $ 
              1298
            </span>
          </div>
          <div>
            <img
              alt="free_shipping"
              class="free_shipping_img"
              data-testid="img-free-shipping-test-id"
              src="ic_shipping.png"
            />
          </div>
        </div>
        <div
          class="cardItem_title "
        >
          <span
            class="title"
          >
            Aperitivo Aperol 750ml
          </span>
        </div>
        <p
          class="fontSize12"
        >
          QUIRINO BEBIDAS
        </p>
      </div>
      <div
        class="mt16 px32 cardItem_sellerState"
      >
        <span
          class="fontSize12"
        >
          Capital Federal
        </span>
      </div>
    </div>
  </div>
  <div
    class="cardItem_border"
  />
</div>
`;
